bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/handlers/VeSyncDeviceOutletHandler.java

```diff
@@ -0,0 +1,248 @@
+import static org.openhab.binding.vesync.internal.dto.requests.VeSyncProtocolConstants.DEVICE_GET_OUTLET_STATUS;
+import static org.openhab.binding.vesync.internal.dto.requests.VeSyncProtocolConstants.DEVICE_SET_SWITCH;
+import static org.openhab.binding.vesync.internal.dto.requests.VeSyncProtocolConstants.MODE_ON;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.vesync.internal.VeSyncBridgeConfiguration;
+import org.openhab.binding.vesync.internal.VeSyncConstants;
+import org.openhab.binding.vesync.internal.dto.requests.VeSyncRequestManagedDeviceBypassV2;
+import org.openhab.binding.vesync.internal.dto.responses.VeSyncV2BypassEnergyHistory;
+import org.openhab.binding.vesync.internal.dto.responses.VeSyncV2BypassEnergyHistory.EnergyHistory.Result.EnergyInfo;
+import org.openhab.binding.vesync.internal.dto.responses.VeSyncV2BypassOutletStatus;
+import org.openhab.core.cache.ExpiringCache;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.MetricPrefix;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VeSyncDeviceOutletHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Goerentz - Add outlets to the supported devices
+ */
+@NonNullByDefault
+public class VeSyncDeviceOutletHandler extends VeSyncBaseDeviceHandler {
+
+    public static final String DEV_TYPE_FAMILY_OUTLET = "OUT";
+
+    public static final int DEFAULT_OUTLET_POLL_RATE = 60;
+
+    public static final String DEV_FAMILY_CORE_WHOG_PLUG = "WHOG";
+    public static final String DEV_FAMILY_CORE_ESW = "ESW";
+    public static final String DEV_FAMILY_CORE_ESWL = "ESWL";
+
+    public static final VeSyncDeviceMetadata COREESW = new VeSyncDeviceMetadata(DEV_FAMILY_CORE_ESW,
+            Collections.emptyList(), Arrays.asList("ESW03-USA", "ESW01-EU", "ESW15-USA"));
+
+    public static final VeSyncDeviceMetadata COREESWL = new VeSyncDeviceMetadata(DEV_FAMILY_CORE_ESWL,
+            Collections.emptyList(), Arrays.asList("ESWL01", "ESWL03"));
+
+    public static final VeSyncDeviceMetadata COREWHOPGPLUG = new VeSyncDeviceMetadata(DEV_FAMILY_CORE_WHOG_PLUG,
+            Arrays.asList("WHOG"), List.of("WHOGPLUG"));
+
+    public static final List<VeSyncDeviceMetadata> SUPPORTED_MODEL_FAMILIES = Arrays.asList(COREESW, COREESWL,
```

Comment:
I think if I've read correctly you've got a one of the WHOG plugs. By the sounds of its this is working with V2 only :) I'd suggest removing the COREESW and COREESWL from the SUPPORTED_MODEL_FAMILIES for now. As on checking PyVeSync they are some of the ones that will need to use some new API's from my PR currently pending, as they have some non V2 standard behaviors. So I suspect they won't play ball unfortunately without a bit more work like the 131 units I have in my PR: https://github.com/webdjoe/pyvesync/blob/2d70bb4c12cdd2c64243aa62a4cd6b0548c99a04/src/pyvesync/vesyncoutlet.py#L182

Reply:
Sorry :(

Reply:
Yes you are correct. I'll wait until you finished your review and will then adjust the Code.

Reply:
I had a read through I'm happy with just the bit above unfortunately.  Nice job :) I think it will just need the previous file mentioned updated and read-me file for when one of the repo maintainers reviews it.

Reply:
Unfortunately, the whogplug is using V1 for the energy consumed within a year. So I would suggest to implement it when your PR has been merged.
Or you could add the needed DTOs to your PR and I will send you the URL payload of the request and of the response.